3.8/aql/fundamentals-limits.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: default
+description: Known limitations that apply in AQL queries
+---
+Known limitations for AQL queries
+=================================
+
+The following limitations are known for AQL queries:
+
+- An AQL query cannot use more than _1000_ result registers. One
+  result register is needed for every named query variable and for 
+  internal/anonymous query variables.
+- An AQL query cannot use more than _2048_ collections/shards.
+- It is not possible to use a collection in a read operation after
+  it was used for a write operation in the same AQL query.
+- In the cluster, all vertex collections and collection that are accessed
+  dynamically via the `DOCUMENT` AQL function must be stated in the 
+  query's initial `WITH` statement.
```

Comment:
I think that _all vertex collections_ might be misleading. Perhaps something like
```suggestion
- In the cluster, all collections that are accessed dynamically must be stated in the 
  query's initial `WITH` statement. These are either accesses via the `DOCUMENT` AQL
  function, or traversals working with collection sets (instead of named graphs).
```
Maybe with links to `DOCUMENT` and https://www.arangodb.com/docs/stable/aql/graphs-traversals.html#working-with-collection-sets.

One other note: We have the `OPTION { vertexCollections }` since 3.7. These used could be used to add the specified collections to the transaction, but I don't know whether they are.